### ISSUE TITLE
is-required doesn't work on checkboxes

### ISSUE THREAD

**Issue by hatched** [canonical-web-and-design/vanilla-framework#3181]
**Describe the bug**

It doesn't appear that the `is-required` class works on checkbox labels.

**To Reproduce**

```jsx
        <div className="row horizonal-rule">
          <div className="col-8 col-start-large-5">
            <input
              type="checkbox"
              id="certificateHasBeenAccepted"
              name="certificateAccepted"
              defaultChecked={false}
              onChange={handleInputChange}
              required="true"
            />
            <label htmlFor="certificateHasBeenAccepted" className="is-required">
              The SSL certificate, if any, has been accepted.
            </label>
          </div>
        </div>
```

**Expected behavior**

I expected the * to show up behind the text to indicate that it was required.

**Screenshots**

![Screen Shot 2020-07-23 at 9 39 17 AM](https://user-images.githubusercontent.com/532033/88306805-6c41d280-ccc8-11ea-9e4a-f890e79768f9.png)
![Screen Shot 2020-07-23 at 9 39 22 AM](https://user-images.githubusercontent.com/532033/88306807-6cda6900-ccc8-11ea-9fce-6d49e1f81b12.png)

Works for text forms with the label in front.

![Screen Shot 2020-07-23 at 9 42 17 AM](https://user-images.githubusercontent.com/532033/88307165-d9edfe80-ccc8-11ea-970c-2778a3d22022.png)

**Desktop (please complete the following information):**

- OS: Catalina
- Browser Firefox
- Version 78.0.2

**Comment by barrymcgee** [comment]
I've dug into this and found it's because of an over-zealous reset style here making opacity on all :after pseudo elements on labels after checkboxes zero.

<img width="277" alt="Screenshot 2020-07-23 at 18 14 06" src="https://user-images.githubusercontent.com/505570/88318872-4b529f00-cd13-11ea-9e38-486b1093b638.png">

Line responsible: https://github.com/canonical-web-and-design/vanilla-framework/blob/master/scss/_base_forms-tick-elements.scss#L146

@lyubomir-popov Was there any particular reason this reset was applied?

**Comment by bartaz** [comment]
@barrymcgee I believe that issue is that both `is-required` and our checkboxes use `::after` on label for rendering.

Checkboxes in Vanilla (if I read the code correctly) use `::before` for checkbox 'box' and `::after` for checkbox tick, so adding `is-required` on top of that conflicts with checkbox styles and will not work itself or affect showing tick (depending on order and specificity of styles).

So, indeed it is a bug, but I can't find an easy fix or workaround from the top of my head. To fix that we will need to change the way either checkboxes or `is-required` uses `::after`...

**Comment by SirSamTumless** [comment]
Going to be fixed by #2931
